frontends/p4/specialize.h

```diff
@@ -113,6 +117,42 @@ class FindSpecializations : public Inspector {
     void postorder(const IR::Declaration_Instance* decl) override;
 };
 
+/** @brief Specializes each Parser and Control by substituting type arguments
+ * and constructor parameters.
+ *
+ * Specifically, each instantiation of each Parser (or Control) with
+ * constructor parameters will cause
+ *  - the creation of a new Parser (Control) with the actual arguments
+ *    substituted for the constructor parameters, and
+ *  - the replacement of the instantiation with a new instantiation using the
+ *    newly-specialized Parser (or Control).
+ *
+ * Note that this pass handles type substition for instantiating generic Parser
+ * or Control types, which is an experimental feature.
+ * 
+ * For example:
+ *
+```
+control c<T>(in T data)(bit<32> size) { ... }
+c<bit<32>>(16) c_inst;
+```
+ *
+ * is converted to
+ *
+```
+control cspec(in bit<32> data) { ... }
+cspec() c_inst;
+```
+ *
```

Comment:
Maybe you can add here that the body of cspec if obtaining from the body of c where T is substituted with bit<32> and size is substituted with 16.